core/templates/pages/exploration-editor-page/history-tab/services/compare-versions.service.spec.ts

```diff
@@ -428,217 +451,180 @@ describe('Compare versions service', function() {
-      var nodeData = null;
       cvs.getDiffGraphData(1, 7).then(function(data) {
-        nodeData = data.nodes;
+        expect(data.nodes).toEqual({
+          1: {
+            newestStateName: 'A',
+            stateProperty: 'changed',
+            originalStateName: 'A'
+          },
+          2: {
+            newestStateName: 'C',
+            stateProperty: 'added',
+            originalStateName: 'B'
+          }
+        });
       });
 
-      $httpBackend.flush();
-      expect(nodeData).toEqual({
-        1: {
-          newestStateName: 'A',
-          stateProperty: 'changed',
-          originalStateName: 'A'
-        },
-        2: {
-          newestStateName: 'C',
-          stateProperty: 'added',
-          originalStateName: 'B'
-        }
-      });
-    });
+      let req1 = httpTestingController.expectOne(
+        '/explorehandler/init/0?v=1');
+      expect(req1.request.method).toEqual('GET');
+      req1.flush(_getStatesData(testExplorationData1[0]));
+
+      let req2 = httpTestingController.expectOne(
+        '/explorehandler/init/0?v=7');
+      expect(req2.request.method).toEqual('GET');
+      req2.flush(_getStatesData(testExplorationData1[6]));
+
+      flushMicrotasks();
+    }));
 
     it('should add new state with same name as old name of renamed state',
-      function() {
-        $httpBackend.expect('GET', '/explorehandler/init/0?v=5')
-          .respond(_getStatesData(testExplorationData1[4]));
-        $httpBackend.expect('GET', '/explorehandler/init/0?v=8')
-          .respond(_getStatesData(testExplorationData1[7]));
+      fakeAsync(() => {
         vts.init(testSnapshots1);
-        var nodeData = null;
         cvs.getDiffGraphData(5, 8).then(function(data) {
-          nodeData = data.nodes;
+          expect(data.nodes).toEqual({
+            1: {
+              newestStateName: 'A',
+              stateProperty: 'unchanged',
+              originalStateName: 'A'
+            },
+            2: {
+              newestStateName: 'C',
+              stateProperty: 'changed',
+              originalStateName: 'B'
+            },
+            3: {
+              newestStateName: 'B',
+              stateProperty: 'added',
+              originalStateName: 'B'
+            }
+          });
         });
-        $httpBackend.flush();
-        expect(nodeData).toEqual({
+
+        let req1 = httpTestingController.expectOne(
+          '/explorehandler/init/0?v=5');
+        expect(req1.request.method).toEqual('GET');
+        req1.flush(_getStatesData(testExplorationData1[4]));
+
+        let req2 = httpTestingController.expectOne(
+          '/explorehandler/init/0?v=8');
+        expect(req2.request.method).toEqual('GET');
+        req2.flush(_getStatesData(testExplorationData1[7]));
+
+        flushMicrotasks();
+      })
+    );
+
+    it('should not include added, then deleted state', fakeAsync(() => {
+      vts.init(testSnapshots1);
+      cvs.getDiffGraphData(7, 9).then(function(data) {
```

Comment:
Use arrow functions here and in other places in thsi file.

Reply:
Will be done when we migrate this particular spec file.